introducing-java/src/main/java/com/onto/javacourse/intro/findingArrayMedian/FindingArrayMedian.java

```diff
@@ -14,40 +14,17 @@ public static int findMedianPosition(int[] array) {
             throw new IllegalArgumentException("Array must not be null or empty");
         }
 
-        int index = 0;
-        int currentLeftSum = 0;
-        int currentRightSum = 0;
+        int totalSum = 0;
+        for (int num : array) {
+            totalSum += num;
+        }
         int leftSum = 0;
-        int rightSum = Integer.MAX_VALUE;
         for (int i = 0; i < array.length; i++) {
-            for (int j = 0; j < i; j++) {
-                if (j != i) {
-                    currentLeftSum += array[j];
-                }
-            }
-            for (int j = array.length - 1; j > i; j--) {
-                if (j != i) {
-                    currentRightSum += array[j];
-                }
-            }
-            if (currentRightSum > currentLeftSum) {
-                if (currentRightSum - currentLeftSum < rightSum - leftSum) {
-                    index = i;
-                    rightSum = currentRightSum;
-                    leftSum = currentLeftSum;
-                    currentRightSum = 0;
-                    currentLeftSum = 0;
-                }
-            } else {
-                if (currentLeftSum - currentRightSum < rightSum - leftSum) {
-                    index = i;
-                    rightSum = currentRightSum;
-                    leftSum = currentLeftSum;
-                    currentRightSum = 0;
-                    currentLeftSum = 0;
-                }
+            leftSum += array[i];
+            if (leftSum >= totalSum / 2) {
```

Comment:
The approach is correct, but we need to find the minimal absolute value of the difference between leftSum and rightSum. Not sure why are you doing this. Am I missing something?